includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -20,7 +20,9 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 			switch ( $type ) {
 				case AutomaticTemplateTypes::TEMPLATE_NAVBOX:
 				case TemplateClassificationService::TEMPLATE_NAVBOX:
-					$text = self::handleNavboxTemplate();
+				case TemplateClassificationService::TEMPLATE_NAV:
+				case TemplateClassificationService::TEMPLATE_FLAG:
+					$text = self::handleHiddenTemplate();
```

Comment:
I am thinking about actually splitting on template purpose (navbox, general navigation, flag) instead of treatment. What do you think?